test/e2e/dns_autoscaling.go

```diff
@@ -207,14 +226,16 @@ func deleteDNSAutoscalerPod(c clientset.Interface) error {
 	return nil
 }
 
-func waitForDNSReplicasSatisfied(c clientset.Interface, expected int, timeout time.Duration) (err error) {
+func waitForDNSReplicasSatisfied(c clientset.Interface, getExpected getExpectReplicasFunc, timeout time.Duration) (err error) {
 	var current int
+	expected := getExpected(c)
 	framework.Logf("Waiting up to %v for kube-dns reach %v replicas", timeout, expected)
 	condition := func() (bool, error) {
 		current, err = getDNSReplicas(c)
 		if err != nil {
 			return false, err
 		}
+		expected = getExpected(c)
 		if current != expected {
```

Comment:
what happens if a node goes NotReady during the test, do we surface the error clearly? I'm guessing the expected count will change because it relists nodes every invocation, and we will have some extra DNS pods in running for ~5m till the master deletes pods on NotReady nodes, so the test will probably time out. Ideally, an error message that tells us a node went notready/un-schedulable during the test would help.